blockball-bukkit-plugin/src/main/java/com/github/shynixn/blockball/bukkit/logic/business/service/BallEntityServiceImpl.kt

```diff
@@ -67,51 +76,72 @@ class BallEntityServiceImpl @Inject constructor(private val pluginProxy: PluginP
         return ballProxy
     }
 
+    /**
+     * Registers entities on the server when not already registered.
+     * Returns true if registered. Returns false when not registered.
+     */
+    override fun registerEntitiesOnServer(): Boolean {
+        if (registered) {
+            return false
+        }
+
+        val slimeClazz = findClazz("com.github.shynixn.blockball.bukkit.logic.business.nms.VERSION.BallHitBox", pluginProxy)
+        val armorStandClazz = findClazz("com.github.shynixn.blockball.bukkit.logic.business.nms.VERSION.BallDesign", pluginProxy)
+
+        entityRegistry.register(slimeClazz, EntityType.SLIME)
+        entityRegistry.register(armorStandClazz, EntityType.ARMORSTAND)
+
+        registered = true
+        return true
+    }
+
     /**
      * Checks the entity collection for invalid ball entities and removes them.
      */
     override fun <E> cleanUpInvalidEntities(entities: Collection<E>) {
         for (entity in entities) {
-            if (entity is ArmorStand && entity.customName != null) {
-                if (entity.customName == "ResourceBallsPlugin") {
-                    val optProxy = findBallFromEntity(entity)
-                    if (!optProxy.isPresent) {
-                        entity.remove()
-
-                        try{
-                            (entity as Any).javaClass.getDeclaredMethod("deleteFromWorld").invoke(entity)
-                        }catch(e : Exception){
-                        }
+            if (entity !is LivingEntity) {
+                continue
+            }
+
+            if (entity.customName == "ResourceBallsPlugin") {
```

Comment:
Check if entity.customName is not null.

Reply:
Like we discussed above, we don't need to check nullability for this